# HG changeset patch
# User Jason R. Coombs <jaraco@jaraco.com>
# Date 1349660887 14400
# Branch cherrypy-3.2.x
# Node ID 01b6adcb3849b2ff4fa31e3298b494f6b136369e
# Parent  9820107d4ffb8058fd507888f90e28c695f6b4c0
Timer class was renamed from _Timer to Timer in Python 3.3. This change adds a compatibility shim to detect this change and reference the base class accordingly. Fixes #1163.

diff -r 9820107d4ffb8058fd507888f90e28c695f6b4c0 -r 01b6adcb3849b2ff4fa31e3298b494f6b136369e cherrypy/_cpcompat.py
--- a/cherrypy/_cpcompat.py	Wed Oct 03 08:02:12 2012 -0400
+++ b/cherrypy/_cpcompat.py	Sun Oct 07 21:48:07 2012 -0400
@@ -18,6 +18,7 @@
 import os
 import re
 import sys
+import threading
 
 if sys.version_info >= (3, 0):
     py3k = True
@@ -325,3 +326,9 @@
     # Python 2
     def next(i):
         return i.next()
+
+if sys.version_info >= (3,3):
+    Timer = threading.Timer
+else:
+    # Python 3.2 and earlier
+    Timer = threading._Timer
diff -r 9820107d4ffb8058fd507888f90e28c695f6b4c0 -r 01b6adcb3849b2ff4fa31e3298b494f6b136369e cherrypy/process/plugins.py
--- a/cherrypy/process/plugins.py	Wed Oct 03 08:02:12 2012 -0400
+++ b/cherrypy/process/plugins.py	Sun Oct 07 21:48:07 2012 -0400
@@ -7,7 +7,7 @@
 import time
 import threading
 
-from cherrypy._cpcompat import basestring, get_daemon, get_thread_ident, ntob, set
+from cherrypy._cpcompat import basestring, get_daemon, get_thread_ident, ntob, set, Timer
 
 # _module__file__base is used by Autoreload to make
 # absolute any filenames retrieved from sys.modules which are not
@@ -421,7 +421,7 @@
             pass
 
 
-class PerpetualTimer(threading._Timer):
+class PerpetualTimer(Timer):
     """A responsive subclass of threading._Timer whose run() method repeats.
 
     Use this timer only when you really need a very interruptible timer;
